src/v/cloud_topics/batcher/batcher.cc

```diff
@@ -249,7 +250,7 @@ ss::future<result<bool>> batcher<Clock>::run_once() noexcept {
         vlog(_logger.error, "Unexpected batcher error: {}", err);
         co_return errc::unexpected_failure;
     }
-    __builtin_unreachable();
+    unreachable();
```

Comment:
I would be way more comfortable with an vassert here instead of undefined behavior. I'd reach for undefined behavior only in hot loops. This one isn't. It is a coroutine, it reaches out to network, etc. We're getting ourselves into dangerous zone for no good reason.

Reply:
Like can you explain why this prints boo 2 times? https://godbolt.org/z/MfqM63hvx

It segv's eventually but what if other code gets into there which doesn't segv and just continues execution? I'm assuming that is possible.

Reply:
will replace this with __builtin_trap or similar